app/services/hashtag_autocomplete_service.rb

```diff
@@ -206,24 +219,20 @@ def search(
 
     return limited_results if limited_results.length >= limit
 
-    # Search the data source for each type, validate and sort results,
-    # and break off from searching more data sources if we reach our limit
+    # Next priority are slugs which start with the search term.
     types_in_priority_order.each do |type|
-      search_results = search_for_type(type, guardian, term, limit - limited_results.length)
-      next if search_results.empty?
-
-      next if !all_data_items_valid?(search_results)
+      limited_results = search_using_condition(limited_results, term, type, limit, SEARCH_CONDITION_STARTS_WITH)
```

Comment:
I think `limit` should be `limit - limited_results.length` because otherwise `limited_results` may have more than `limit` results and you return that without limiting to `limit` elements (unless it gets to the last line of the line, which might not matter very much).

Reply:
Yes you are right, good catch!

Reply:
Actually no `search_using_condition` already does this inside